components/content-service/pkg/archive/tar.go

```diff
@@ -69,9 +75,86 @@ func WithGIDMapping(mappings []IDMapping) BuildTarbalOption {
 	}
 }
 
+// ExtractTarbal extracts an OCI compatible tar file src to the folder dst, expecting the overlay whiteout format
+func ExtractTarbal(ctx context.Context, src io.Reader, dst string, opts ...TarOption) (err error) {
+	var cfg tarConfig
+	start := time.Now()
+	for _, opt := range opts {
+		opt(&cfg)
+	}
+
+	//nolint:staticcheck,ineffassign
+	span, ctx := opentracing.StartSpanFromContext(ctx, "extractTarbal")
+	span.LogKV("src", src, "dst", dst)
+	defer tracing.FinishSpan(span, &err)
+
+	re, wr := io.Pipe()
+	src = io.TeeReader(src, wr)
+	tarReader := tar.NewReader(re)
+	type Info struct {
+		UID, GID int
+	}
+	finished := make(chan bool)
+	m := make(map[string]Info)
+	go func() {
+		for {
+			hdr, err := tarReader.Next()
+			if err == io.EOF {
+				finished <- true
+				return
+			}
+			m[hdr.Name] = Info{
+				UID: hdr.Uid,
+				GID: hdr.Gid,
+			}
+		}
+	}()
+
+	tarcmd := exec.Command("tar", "x")
+	tarcmd.Dir = dst
+	tarcmd.Stdin = src
+
+	msg, err := tarcmd.CombinedOutput()
+	if err != nil {
+		return xerrors.Errorf("tar %s: %s", dst, err.Error()+";"+string(msg))
+	}
+	<-finished
+
+	// lets create a sorted list of pathes and chown depth first.
+	pathes := make([]string, len(m))
+	i := 0
+	for path := range m {
+		pathes[i] = path
+		i++
+	}
+	sort.Sort(sort.Reverse(sort.StringSlice(pathes)))
+	for _, p := range pathes {
+		v := m[p]
+		uid := toHostId(v.UID, cfg.UIDMaps)
+		gid := toHostId(v.GID, cfg.GIDMaps)
+		err = os.Lchown(path.Join(dst, p), uid, gid)
+		if err != nil {
+			log.Errorf("cannot chown %s to %d:%d : %s", p, uid, gid, err.Error())
+		}
+	}
+	t := time.Now()
+	log.Debugf("Untar took %d ms", t.Sub(start).Milliseconds())
```

Comment:
```suggestion
	log.WithField("duration", time.Since(start).Milliseconds()).Debug("untar complete")
```

better yet, take the untar durations from traces or metrics rather than logs.